pkg/virtual/framework/wrappers/rbac/merging.go

```diff
@@ -0,0 +1,469 @@
+	panic("not implemented")
+}
+
+func (i *mergedClusterRoleBindingInformer) Lister() rbaclisters.ClusterRoleBindingLister {
+	aggregatedListers := make([]rbaclisters.ClusterRoleBindingLister, 0)
+	for _, inf := range i.informers {
+		aggregatedListers = append(aggregatedListers, inf.Lister())
+	}
+	return &mergedClusterRoleBindingLister{
+		listers: aggregatedListers,
+	}
+}
+
+func (l *mergedClusterRoleBindingLister) List(selector labels.Selector) (ret []*rbacv1.ClusterRoleBinding, err error) {
+	aggregatedList := make([]*rbacv1.ClusterRoleBinding, 0)
+	for _, lister := range l.listers {
+		list, err := lister.List(selector)
+		if err != nil {
+			return nil, err
+		}
+		aggregatedList = append(aggregatedList, list...)
+	}
+	return aggregatedList, nil
+}
+
+func (l *mergedClusterRoleBindingLister) Get(name string) (*rbacv1.ClusterRoleBinding, error) {
+	var errorHolder error
+	for _, lister := range l.listers {
+		item, err := lister.Get(name)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				errorHolder = err
+			}
+		} else {
+			return item, nil
+		}
+	}
+	return nil, errorHolder
+}
+
+func (l *mergedClusterRoleBindingLister) ListWithContext(ctx context.Context, selector labels.Selector) (ret []*rbacv1.ClusterRoleBinding, err error) {
+	aggregatedList := make([]*rbacv1.ClusterRoleBinding, 0)
+	for _, lister := range l.listers {
+		list, err := lister.ListWithContext(ctx, selector)
+		if err != nil {
+			return nil, err
+		}
+		aggregatedList = append(aggregatedList, list...)
+	}
+	return aggregatedList, nil
+}
+
+func (l *mergedClusterRoleBindingLister) GetWithContext(ctx context.Context, name string) (*rbacv1.ClusterRoleBinding, error) {
+	var errorHolder error
+	for _, lister := range l.listers {
+		item, err := lister.GetWithContext(ctx, name)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				errorHolder = err
+			}
+		} else {
+			return item, nil
+		}
+	}
+	return nil, errorHolder
+}
+
+func MergedClusterRoleInformer(informers []rbacinformers.ClusterRoleInformer) rbacinformers.ClusterRoleInformer {
+	return &mergedClusterRoleInformer{
+		informers: informers,
+	}
+}
+
+var _ rbacinformers.ClusterRoleInformer = (*mergedClusterRoleInformer)(nil)
+var _ rbaclisters.ClusterRoleLister = (*mergedClusterRoleLister)(nil)
+
+type mergedClusterRoleInformer struct {
+	informers []rbacinformers.ClusterRoleInformer
+}
+
+type mergedClusterRoleLister struct {
+	listers []rbaclisters.ClusterRoleLister
+}
+
+func (i *mergedClusterRoleInformer) Informer() cache.SharedIndexInformer {
+	panic("not implemented")
+}
+
+func (i *mergedClusterRoleInformer) Lister() rbaclisters.ClusterRoleLister {
+	aggregatedListers := make([]rbaclisters.ClusterRoleLister, 0)
+	for _, inf := range i.informers {
+		aggregatedListers = append(aggregatedListers, inf.Lister())
+	}
+	return &mergedClusterRoleLister{
+		listers: aggregatedListers,
+	}
+}
+
+func (l *mergedClusterRoleLister) List(selector labels.Selector) (ret []*rbacv1.ClusterRole, err error) {
```

Comment:
either we should also merge each entry, or we should panic or something to make sure this is never called. After all it has wrong semantics.

Reply:
Let's merge them. The code will be there in the next push.